pysteps/io/importers.py

```diff
@@ -131,6 +133,230 @@
 except ImportError:
     PYPROJ_IMPORTED = False
 
+try:
+    import pygrib
+
+    PYGRIB_IMPORTED = True
+except ImportError:
+    PYGRIB_IMPORTED = False
+
+
+def _check_coords_range(selected_range, coordinate, full_range):
+    """Check that the coordinates range arguments follow the expected pattern in the **import_mrms** function."""
+
+    if selected_range is None:
+        return sorted(full_range)
+
+    if not isinstance(selected_range, (list, tuple)):
+
+        if len(selected_range) != 2:
+            raise ValueError(f"The {coordinate} range must be None or a two-element tuple or list")
+
+        selected_range = list(selected_range)  # Make mutable
+
+        for i in range(2):
+            if selected_range[i] is None:
+                selected_range[i] = full_range
+
+        selected_range.sort()
+
+    return tuple(selected_range)
+
+
+def import_mrms(filename, fillna=np.nan, lat_range=None, lon_range=None,
+                dtype='float32', block_size=4, **kwargs):
+    """
+    Importer for NSSL's Multi-Radar/Multi-Sensor System
+    ([MRMS](https://www.nssl.noaa.gov/projects/mrms/)) rainrate product
+    (grib format).
+
+    The rainrate values are expressed in mm/h, and the dimensions of the data
+    array are [latitude, longitude]. The first grid point (0,0) corresponds to
+    the upper left corner of the domain, while (last i, last j) denote the
+    lower right corner.
+
+    Due to the large size of the dataset (3500 x 7000), a float32 type is used
+    by default to reduce the memory footprint. However, be aware that when this
+    array is passed to a pystep function, it may be converted to double
+    precision, doubling the memory footprint.
+    To change the precision of the data, use the *dtype* keyword.
+
+    Also, by default, the original data is downscaled by 4
+    (that gives a 4 km grid spacing).
+    In case that the original grid spacing is needed, use `block_size=1`.
+    But be aware that a single composite in double precipitation will
+    require 186 Mb of memory.
+
+    Finally, the precipitation data can only be extracted sub region of the
+    full domain using the `lat_range` and `lon_range` keywords.
+    By default, the entire domain is returned.
+
+    Parameters
+    ----------
+
+    filename : str
+        Name of the file to import.
+
+    fillna : float or np.nan
+        Value used to represent the missing data ("No Coverage").
+        By default, np.nan is used.
+
+    dtype : str
+        Data-type to which the array is cast.
+        Valid values:  "float32", "float64", "single", and "double".
+
```

Comment:
The `lat_range` and `lon_range` arguments have somewhat the same functionality as the `extent` argument in the `import_saf_crri` method. If possible, I suggest using the same approach for both methods to provide some consistency among the importers.